libnavigation-util/src/main/java/com/mapbox/navigation/utils/internal/LoggerFrontend.kt

```diff
@@ -10,48 +12,31 @@ interface LoggerFrontend {
     fun logW(msg: String, category: String? = null)
 }
 
-internal class NoLoggingFrontend : LoggerFrontend {
-    override fun logV(msg: String, category: String?) {
-    }
-
-    override fun logD(msg: String, category: String?) {
-    }
-
-    override fun logI(msg: String, category: String?) {
-    }
-
-    override fun logE(msg: String, category: String?) {
-    }
-
-    override fun logW(msg: String, category: String?) {
-    }
-}
-
```

Comment:
Does unit tests work without `NoLoggingFrontend`? I expect them to fail 🤔

Reply:
I added it as a default logger implementation for easy unit testing. You was test any class without any logger setup.

Reply:
Possibly, I'll have to address that. I still do think that it's a better option to mock/replace the logger in tests rather than initialize with an empty logger because the valid logger was provided only in the `MapboxNavigation` and there are components of Nav SDK that could want to log before `MapboxNavigation` is created (any UI components or util classes for example).

Reply:
> there are components of Nav SDK that could want to log before MapboxNavigation is created (any UI components or util classes for example).

That's a fair point! I didn't consider this case 😱 

Please be aware, that if you do static mocking for logger in all tests, it will slow down tests execution by ~2 minutes. Replacing logger in tests by fake implementation is a faster option.

Reply:
Done, ready for another round.